sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -73,6 +76,26 @@ protected TestEnvironment()
             _type = GetType();
 
             ParseEnvironmentFile();
+
+            string tenantId = GetOptionalVariable("TENANT_ID");
+            string clientId = GetOptionalVariable("CLIENT_ID");
+            string clientSecret = GetOptionalVariable("CLIENT_SECRET");
+            string authorityHost = GetOptionalVariable("AZURE_AUTHORITY_HOST");
+
+            // configure pipeline if credential vars are available
+            if (tenantId != null && clientId != null && clientSecret != null && authorityHost != null)
```

Comment:
It may not be safe to cache the credential, and therefore the pipeline because different environments may have different credentials available.

Reply:
Removed the pipeline/credential caching - it will be recreated once per Environment type.